rc2024 build fails on macOS 12 (LLVM 14), missing `#include <unordered_map>`

I was updating `librcsc` package in Homebrew (https://github.com/Homebrew/homebrew-core/pull/180809) and noticed the build failed on macOS 12 Monterey (Xcode 14.2 / LLVM 14.0.0).

This appears to be due to missing include. Some LLVM/GCC versions seem to indirectly pull in header given CI passes on Ubuntu 22.04 (GCC 11) and macOS 13+ (LLVM 15+).

https://github.com/Homebrew/homebrew-core/actions/runs/10334715364/job/28611025543#step:3:769
```
 parser_simdjson.cpp:54:10: error: no template named 'unordered_map' in namespace 'std'
      std::unordered_map< std::string, Func > funcs_;
      ~~~~~^
  parser_simdjson.cpp:150:10: error: no member named 'unordered_map' in namespace 'std'
      std::unordered_map< std::string, Func >::iterator it = funcs_.find( key );
      ~~~~~^
  parser_simdjson.cpp:150:36: error: expected '(' for function-style cast or type construction
      std::unordered_map< std::string, Func >::iterator it = funcs_.find( key );
                          ~~~~~~~~~~~^
  parser_simdjson.cpp:151:10: error: use of undeclared identifier 'it'
      if ( it == funcs_.end() )
           ^
  parser_simdjson.cpp:158:12: error: use of undeclared identifier 'it'
      if ( ! it->second( field.value(), handler ) )
             ^
```

--- a/rcsc/rcg/parser_simdjson.cpp
+++ b/rcsc/rcg/parser_simdjson.cpp
@@ -41,6 +41,7 @@
 
 #include "simdjson/simdjson.h"
 
+#include <unordered_map>
 #include <string_view>
 #include <functional>
 

